src/interface/src/app/signup/signup.component.ts

```diff
@@ -52,11 +58,40 @@ export class SignupComponent {
     this.authService.resendValidationEmail(email).subscribe();
   }
 
+  getEmailError() {
+    var emailErrors = this.form.controls['email'].errors;
+    if (emailErrors && 'required' in emailErrors) {
+      return 'Email is required.';
+    }
+    if (emailErrors && 'pattern' in emailErrors) {
+      return 'Email must be the correct format.';
+    }
+    if (emailErrors && 'accountExists' in emailErrors) {
+      return 'An account with this email already exists.';
+    }
+    return 'Unknown error.';
+  }
+
+  getFormErrors(): string {
+    if (this.form.errors && 'newPasswordsMustMatch' in this.form.errors) {
```

Comment:
maybe you can do something like `if(!this.form.errors) return` to avoid checking for this each time.
Same for the one above with `emailErrors`